site-src/guides/multiple-ns.md

```diff
@@ -1,38 +1,65 @@
+
+- The site team has two applications, home and login, that are running behind `foo.example.com`. They have decided to deploy independent HTTPRoutes for each application so that the routing can be configured indendently for each app. By having an independent HTTPRoute for each app, it allows them to update the canary weights for the login app without having to touch any routing configuration for home.
+- The store team has a single Service called `store` that they have deployed in the `store` Namespace. 
+- The Foobar Corporation operates behind the `foo.example.com` domain so they would like to host all applications on the same Gateway resource. This is controlled by a central infrastructure team, operating in the `infra` Namespace.
+- Lastly, the security team controls the certificate for `foo.example.com`. By managing this certificate through the single shared Gateway they are able to centrally control security without directly involving application teams.
+
+The logical relationship between the Gateway API resources looks like this:
+
+![Cross-Namespace routing](/images/cross-namespace-routing.svg)
+
+## Cross-namespace Route binding
+
+[Route binding](/concepts/api-overview/#route-binding) is an important concept that dictates how routes and Gateways select eachother for binding. It is especially relevant when there are multiple Gateways and multiple Namespaces in a cluster. Route binding is based on the principle of bi-directional selection - this means that each resource, the Gateway and the Route, have independent control to determine which resources they permit binding with. This allows Route-owners and Gateway-owners to be independent actors. Route-owners can specify that they will bind with all Gateways in the cluster, or only Gateways from a specific Namespace, with a specific label selector, or an individual Gateway. Similarly, Gateways provide the same level of control. This allows a cluster to be more self-governed, which requires less central administration to ensure that Routes are not over-exposed.
+
+## Resource Deployment
+
+The infrastructure team deploys the `shared-gateway` Gateway into the `infra` Namespace. 
+
+```yaml
+{% include 'cross-namespace-routing/gateway.yaml' %}
 ```
-{% include 'routes-in-multiple-namespaces.yaml' %}
+
+A couple notes about this Gateway:
+
+- It is matching for the `foo.example.com` domain. This is configured on the Gateway so that each HTTPRoute does not also have to configure hostname matching, since they are all using the same domain.
+- The Gateway is configured for HTTPS and references the `foo-example-com` Secret. This allows the certificate to be managed centrally for all applications which are using this Gateway.
+- `routes.Namespaces.from: All` is the configuration to allow routes from any Namespace to use this Gateway. This impacts how routes are able to bind to this Gateway.
```

Comment:
This is fairly insecure... ideally we would have route delegation here but since that is not implemented, should we at least explicitly select `store` and `site` namespaces so `attacker` namespace cannot attach some random route?

Reply:
We should say this what the Ingress controllers do as part of the standard install, but point out that it's not very secure.

Reply:
Okay good point, I could use the Namespace label selector on the Gateway. It does make this example more complicated because I have to go through the fact that Namespaces must be labelled. @robscott I think it's probably a good thing that the new Gateway->Route permissions allow it to reference Namespaces directly by name.

Reply:
All namespaces are labeled with `kubernetes.io/metadata.name` in recent Kubernetes versions. I think we should encourage this label, as all other namespace labels are not enforceable - Kubernetes will guarantee `metadata.name` == the name of the namespace.

I think https://github.com/istio/istio/blob/58867c37948598b2a03e23fa451a8211d7dad18b/pilot/pkg/config/kube/gateway/testdata/delegated.yaml#L1 is the best practice, although its for delegating domains not paths (which is more secure)

Reply:
@howardjohn I agree that if dealing with untrusted namespaces, it should be done a little more explicitly. I realized though that for this use-case (single domain, multiple paths in different namespaces), it actually doesn't work because the Namespace label selector is an AND and not a OR (I think). I would need to be able to select multiple Namespaces from a single listener.

I am leaving this as-is until the new route binding proposal is merged in in v1alpha2. Then it will be much easier for a single listener to select multiple Namespaces cc @robscott

Reply:
```
    - key: kubernetes.io/metadata.name
      operator: In
      values:
      - ns1
      - ns2
```

I think you can do something like that to get OR ^. But fine to do later